src/App.tsx

```diff
@@ -1,93 +1,68 @@
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useState } from 'react';
+import { TodosContext } from './components/TodosContext';
+import { Todo } from './types/Todo';
+import { TodoContent } from './components/TodoContent';
+import { useLocalStorage } from './hooks/useLocalStorage';
+import { TodoFooter } from './components/TodoFooter';
+import { TodosFilter } from './types/TodosFilter';
 
 export const App: React.FC = () => {
+  const [query, setQuery] = useState('');
+  const [todos, setTodos] = useLocalStorage<Todo[]>('todos', []);
+  const [todosFilter, setTodosFilter] = useState<TodosFilter>(TodosFilter.all);
+  const [todoEditId, setTodoEditId] = useState<number>(0);
+  const [todoEdit, setTodoEdit] = useState<string>('');
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    if (query.trim()) {
+      const todo = {
+        id: Date.now(),
+        title: query,
+        completed: false,
+      };
+
+      setTodos([...todos, todo]);
```

Comment:
```suggestion
      setTodos(prevTodos => [...prevTodos, todo]);
```
It is safer to pass updater callback when new state values depends on previous one